### PR TITLE
ContextFactory reused to initialize language-server context

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Move `ContextFactory` to common module, so it is also available in `language-server` module and can be used there as well. As a result the [initialization of experimental Espresso support](https://github.com/enso-org/enso/pull/10670#discussion_r1692714033) is now unified. Otherwise this PR is _just a refactoring_ except a single change: when using `--inspect` and `--run` hitting `Standard.Base.Runtime.Debug.breakpoint` statement, the debugger stops in _Chrome Dev Tools_, as the internal debug server is disabled.

### Checklist

- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
- [x] Unit tests continue to work

### CODE REVIEW

File: engine/common/src/main/java/org/enso/common/ContextFactory.java [hunk 1]
```diff
@@ -197,7 +196,7 @@ PolyglotContext build() {
           .option("java.UseBindingsLoader", "true")
```
Comment (JaroslavTulach) [inline engine/common/src/main/java/org/enso/common/ContextFactory.java:196]: This `if (ENGINE_HAS_JAVA)` block is now shared between CLI and language-server.

File: engine/language-server/src/main/java/org/enso/languageserver/boot/resource/TruffleContextInitialization.java [hunk 2]
```diff
@@ -24,34 +23,24 @@ public class TruffleContextInitialization extends LockedInitialization {
...
-    this.truffleContextBuilder = truffleContextBuilder;
+    this.contextFactory = factory;
     this.supervisor = supervisor;
     this.eventStream = eventStream;
   }
 
   @Override
   public void initComponent() {
     logger.trace("Creating Runtime context");
-    if (Engine.newBuilder()
```
Comment (JaroslavTulach) [inline engine/language-server/src/main/java/org/enso/languageserver/boot/resource/TruffleContextInitialization.java:43]: This Espresso block was inconsistent with https://github.com/enso-org/enso/pull/10670/files#r1692712349 - now the handling of Espresso is the same in language-server as well as CLI.

File: engine/runner/src/main/java/org/enso/runner/Main.java [hunk 3]
```diff
@@ -706,8 +705,16 @@ private void run(
             .enableStaticAnalysis(enableStaticAnalysis)
             .executionEnvironment(executionEnvironment != null ? executionEnvironment : "live")
             .warningsLimit(warningsLimit)
-            .options(options)
-            .build();
+            .options(options);
+
+    if (inspect) {
+      options.put("inspect", "");
```
Comment (JaroslavTulach) [inline engine/runner/src/main/java/org/enso/runner/Main.java:711]: I realized that when one uses `--inspect` and hit `Debug.breakpoint` statement we actually _hit the breakpoint twice_. Once in Chrome Dev Tools and once in REPL. I believe it is better to have just a single debugger running. As such I am proposing to **disable debug server** when `--inspect` option is used.

Comment (radeusgd) [reply engine/runner/src/main/java/org/enso/runner/Main.java:711]: That definitely makes sense.

**Review by Akirathan — APPROVED** [review]

File: engine/common/src/main/java/org/enso/common/RuntimeOptions.java [hunk 4]
```diff
@@ -124,19 +123,13 @@ public class RuntimeOptions {
 
   public static final String ENABLE_EXECUTION_TIMER = optionName("enableExecutionTimer");
 
-  @Option(
```
Comment (radeusgd) [inline engine/common/src/main/java/org/enso/common/RuntimeOptions.java:127]: Why are these annotations with _seemingly helpful descriptions_ removed but options themselves kept?

Comment (JaroslavTulach) [reply engine/common/src/main/java/org/enso/common/RuntimeOptions.java:127]: I concluded (to myself) that nobody can be using the `@Option` annotation values. 

The annotation is `com.oracle.truffle.api.Option`, but `RuntimeOptions` are being used from `runner.jar` and `runner.jar` doesn't have dependency on Truffle and it shouldn't have.

No test failed anywhere.

**Review by radeusgd — APPROVED** [review]
Looks good

File: engine/common/src/main/java/org/enso/common/RuntimeOptions.java (hunk 4 above)
Comment (radeusgd) [reply engine/common/src/main/java/org/enso/common/RuntimeOptions.java:127]: I see. Sounds fair to remove them, although I'd prefer to keep the `help` text as a javadoc comment for one of the fields - I find these texts quite helpful in understanding what each option stands for.

Comment (JaroslavTulach) [reply engine/common/src/main/java/org/enso/common/RuntimeOptions.java:127]: I can keep them. a238f0ec59. Somebody should however write documentation for the other options.

File: engine/runner-common/src/main/scala/org/enso/runner/common/CompilerBasedDependencyExtractor.scala [hunk 5]
```diff
@@ -55,18 +52,10 @@ class CompilerBasedDependencyExtractor(logLevel: Level)
     * project root.
     */
   private def createContextWithProject(pkg: Package[File]): PolyglotContext = {
-    val context = Context
-      .newBuilder()
-      .allowExperimentalOptions(true)
-      .allowAllAccess(true)
-      .allowHostAccess(new HostAccessFactory().allWithTypeMapping())
-      .option(RuntimeOptions.PROJECT_ROOT, pkg.root.getCanonicalPath)
-      .option("js.foreign-object-prototype", "true")
```
Comment (hubertp) [inline engine/runner-common/src/main/scala/org/enso/runner/common/CompilerBasedDependencyExtractor.scala:64]: This option is no longer needed?

**Review by hubertp — APPROVED** [review]

File: engine/runner-common/src/main/scala/org/enso/runner/common/CompilerBasedDependencyExtractor.scala (hunk 5 above)
Comment (JaroslavTulach) [reply engine/runner-common/src/main/scala/org/enso/runner/common/CompilerBasedDependencyExtractor.scala:64]: I don't know what that option is good for?

 In any case, it is not used by CLI. I see two PRs that refer to that option:
- #7991
- #7855

There were no failed tests when the option was removed, so it is not essential for any use-case we test for.

**Pull request merged**